pulser/sequence.py

```diff
@@ -479,6 +543,9 @@ def add(
         if self.is_parametrized():
             if not isinstance(pulse, Parametrized):
                 self._validate_pulse(pulse, channel)
+            # Sequence is marked as non-empty on the first added pulse
+            if self._empty_sequence:
+                self._empty_sequence = False
```

Comment:
If I understand correctly, this is needed  because the magnetic field can be set only on an empty sequence. But I don't understand why the empty sequence constraint is necessary

Reply:
You mean the `if`, right? It's not necessary, actually. I just thought it would be a bit cleaner and faster, which is true but completely negligible. Would you advocate for changing it to just setting `self._empty_sequence = False` every time?

Reply:
Ah sorry, I didn't explain myself very well. I agree with the `if`, I wouldn't change it. I was just wondering why we can only set the magnetic field on an empty sequence

Reply:
Slimane had the same question just now so I'll quote my reply:

> It's just because we wanted it to be clear that it is set before the sequence begins and cannot be changed. If you could do it at the middle of a sequence, it might give the impression that it only turns on then